web3/contract.py

```diff
@@ -820,3 +871,27 @@ def estimate_gas_for_function(contract=None,
 
     gas_estimate = contract.web3.eth.estimateGas(estimate_transaction)
     return gas_estimate
+
+
+def prepare_transaction_for_function(contract=None,
+                                     function_name=None,
+                                     transaction=None,
+                                     *args,
+                                     **kwargs):
+    """Prepares a dictionary with the fields required to make the given transaction
+
+    Don't call this directly, instead use :meth:`Contract.prepareTransaction`
+    on your contract instance.
+    """
+    prepared_transaction = contract._prepare_transaction(
+        fn_name=function_name,
+        fn_args=args,
+        fn_kwargs=kwargs,
+        transaction=transaction,
+    )
+
+    prepared_transaction = fill_transaction_defaults(contract.web3, prepared_transaction)
+    # from is not a valid transaction part and used only for retreiving nonce
+    prepared_transaction.pop('from')
```

Comment:
If it does throw an exception (seems plausible), it might be reasonable to actually alter the `eth.account` code to only throw an exception if `from` is different from the key that's signing the transaction. If you confirm the exception, but aren't comfortable making that change, let me know and I'm happy to add it in.

Reply:
Ok i definitely made that up 😄  Anyway with the removal of nonce defaulting, `from` won't have to be passed into this method anyway.